e2e/kind.sh

```
@@ -0,0 +1,112 @@
+		if [[ $? == 0 && $ping == "pong" ]]; then
+			break
+		else
+			echo "expected \"pong\" got \"$ping\""
+			sleep 5
+		fi
+	done
+}
+
+check_adjacent() {
+	echo
+	echo "$($KUBECTL_BINARY get pods -l app.kubernetes.io/name=curl -o name | xargs -i $KUBECTL_BINARY exec {} -- /bin/sh -c 'curl -s adjacency:8080/?format=fancy')"
+	assert_equals "12" \
+		"$($KUBECTL_BINARY get pods -l app.kubernetes.io/name=curl -o name | xargs -i $KUBECTL_BINARY exec {} -- /bin/sh -c 'curl -s adjacency:8080/?format=json' | jq | grep -c true)" \
+		"Adjacency returned the wrong number of successful pings"
+	echo "sleep for 30s (one reconciliation period) and try again..."
+	sleep 30
+	echo
+	echo "$($KUBECTL_BINARY get pods -l app.kubernetes.io/name=curl -o name | xargs -i $KUBECTL_BINARY exec {} -- /bin/sh -c 'curl -s adjacency:8080/?format=fancy')"
+	assert_equals "12" \
+		"$($KUBECTL_BINARY get pods -l app.kubernetes.io/name=curl -o name | xargs -i $KUBECTL_BINARY exec {} -- /bin/sh -c 'curl -s adjacency:8080/?format=json' | jq | grep -c true)" \
+		 "Adjacency returned the wrong number of successful pings"
+}
+
+test_locationmesh() {
+	$KUBECTL_BINARY patch ds -n kube-system kilo -p '{"spec": {"template":{"spec":{"containers":[{"name":"kilo","args":["--hostname=$(NODE_NAME)","--create-interface=false","--mesh-granularity=location"]}]}}}}'
+	sleep 5
+	block_until_ready_by_name kube-system kilo-userspace 
+	block_until_ping
+	sleep 30
+	check_adjacent
+}
+
+test_fullmesh() {
+	$KUBECTL_BINARY patch ds -n kube-system kilo -p '{"spec": {"template":{"spec":{"containers":[{"name":"kilo","args":["--hostname=$(NODE_NAME)","--create-interface=false","--mesh-granularity=full"]}]}}}}'
+	sleep 5
+	block_until_ready_by_name kube-system kilo-userspace 
+	block_until_ping
+	sleep 30
+	check_adjacent
+}
+
+teardown_suite () {
+	$KIND_BINARY delete clusters $KIND_CLUSTER
+	rm $KUBECONFIG
```

Looks like this is failing. Should we make it "rm -f" to ensure that failures are ignored? Or can we skip this entirely since the file seems to not exist?

---

This should not effect the exit code. Apparently kind delete cluster also deletes the kubeconfig?

---

In any case this line can be deleted.